libdnf/rpm/solv_sack.cpp

```diff
@@ -0,0 +1,456 @@
+                REPO_EXTEND_SOLVABLES | REPO_LOCALPOOL,
+                [](LibsolvRepo * repo, FILE * fp) { return repo_add_rpmmd(repo, fp, 0, REPO_EXTEND_SOLVABLES) == 0; });
+            repo_impl->libsolv_repo_ext.state_other = retval.state;
+            repo_impl->libsolv_repo_ext.other_repodata = retval.solv_id;
+        } catch (const NoCapability & ex) {
+            logger.debug(fmt::format("no other metadata available for {}", repo_impl->id));
+        }
+        if (repo_impl->libsolv_repo_ext.state_other == LibsolvRepoExt::DataState::LOADED_FETCH && build_cache) {
+            pImpl->write_ext(repo_impl->libsolv_repo_ext, LibsolvRepoExt::DataType::OTHER, SOLV_EXT_OTHER);
+        }
+    }
+    if (any(flags & LoadRepoFlags::USE_PRESTO)) {
+        try {
+            auto retval = pImpl->load_repo_ext(
+                repo,
+                SOLV_EXT_PRESTO,
+                Repo::Impl::MD_FILENAME_PRESTODELTA,
+                REPO_EXTEND_SOLVABLES,
+                [](LibsolvRepo * repo, FILE * fp) { return repo_add_deltainfoxml(repo, fp, 0) == 0; });
+            repo_impl->libsolv_repo_ext.state_presto = retval.state;
+            repo_impl->libsolv_repo_ext.presto_repodata = retval.solv_id;
+        } catch (const NoCapability & ex) {
+            logger.debug(fmt::format("no presto metadata available for {}", repo_impl->id));
+        }
+        if (repo_impl->libsolv_repo_ext.state_presto == LibsolvRepoExt::DataState::LOADED_FETCH && build_cache) {
+            pImpl->write_ext(repo_impl->libsolv_repo_ext, LibsolvRepoExt::DataType::PRESTO, SOLV_EXT_PRESTO);
+        }
+    }
+    // updateinfo must come *after* all other extensions, as it is not a real
+    //   extension, but contains a new set of packages
+    if (any(flags & LoadRepoFlags::USE_UPDATEINFO)) {
+        try {
+            // the updateinfo is not a real extension flags = 0
+            auto retval = pImpl->load_repo_ext(
+                repo, SOLV_EXT_UPDATEINFO, Repo::Impl::MD_FILENAME_UPDATEINFO, 0, [](LibsolvRepo * repo, FILE * fp) {
+                    return repo_add_updateinfoxml(repo, fp, 0) == 0;
+                });
+            repo_impl->libsolv_repo_ext.state_updateinfo = retval.state;
+            repo_impl->libsolv_repo_ext.updateinfo_repodata = retval.solv_id;
+        } catch (const NoCapability & ex) {
+            logger.debug(fmt::format("no updateinfo available for {}", repo_impl->id));
+        }
+        if (repo_impl->libsolv_repo_ext.state_updateinfo == LibsolvRepoExt::DataState::LOADED_FETCH && build_cache) {
+            pImpl->write_ext(repo_impl->libsolv_repo_ext, LibsolvRepoExt::DataType::UPDATEINFO, SOLV_EXT_UPDATEINFO);
+        }
+    }
+    pImpl->considered_uptodate = false;
+}
+
+std::string SolvSack::Impl::give_repo_solv_cache_fn(const std::string & repoid, const char * ext) {
+    auto & cachedir = base->get_config().cachedir().get_value();
+    std::string fn = cachedir + "/" + repoid;
```

Comment:
What about when cachedir == '/tmp/'

Reply:
it will create a path /tmp//<repoid>

Reply:
I think that we planned to store solv files in repo cache directory.

Reply:
I don't understand the question? You are meaning doubling of `/` character. That's not problem. But fact is that I plan use C++17  `std::filesystem::path` in future. So I can do it now.